pkg/controller/reconciler.go

```diff
@@ -69,22 +69,22 @@ func (r *reconciler) InjectAPIReader(reader client.Reader) error {
 
 func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	obj := r.typ.DeepCopyObject().(client.Object)
+	obj.SetName(request.Name)
+	obj.SetNamespace(request.Namespace)
+	logger := r.logger.WithValues("name", obj.GetName(), "namespace", obj.GetNamespace())
+
 	if err := r.client.Get(ctx, request.NamespacedName, obj); err != nil {
 		if apierrors.IsNotFound(err) {
+			if r.shouldEnsureDeleted {
```

Comment:
Do we really need to trigger a double deletion in case the api returns that the object does not exists?

If yes, should we then not rather try it always instead of making in configurable?

Reply:
See the remarks above. This is a safety mechanism that I introduced because we observed "orphan" virtualmachines in real clusters (and the same could happen to publicipaddresses). How did they appear in the first place is rather a mystery (I suspect manual intervention), but this mechanism will effectively get rid of them.